sapphire/mysqlConnector/Connection.cpp

```diff
@@ -222,6 +234,47 @@ MYSQL* Mysql::Connection::getRawCon()
    return m_pRawCon;
 }
 
+void Mysql::Connection::pingLoop()
+{
+   // todo: this sleep is so dumb, initialise the loop somewhere outside the constructor
+   std::this_thread::sleep_for( std::chrono::seconds( 5 ) );
+   std::chrono::seconds timeout;
+   {
+      std::lock_guard< std::mutex > lock( m_connMutex );
+
+      if( isClosed() )
+         throw std::runtime_error( "Mysql: No connection to retrieve wait_timeout!" );
+
+      std::string query( "SHOW VARIABLES LIKE 'wait_timeout';" );
+      auto res = mysql_real_query( m_pRawCon, query.c_str(), query.length() );
+
+      if( res != 0 )
+         throw std::runtime_error("Mysql::pingLoop wait_timeout query failed!");
+
+      auto pRes = mysql_store_result( m_pRawCon );
+      auto row = mysql_fetch_row( pRes );
+
+      // todo: probably use a config value for ping timeout and use that instead of hardcoded 30sec leeway
+      int seconds = atoi( row[1] );
+      seconds = seconds - 30 < 1 ? 30 : seconds - 30;
+
+      timeout = std::chrono::seconds( seconds );
+      mysql_free_result( pRes );
+   }
+
+   while( true )
```

Comment:
Give that gremlin some space to live in:

```cpp
while ( true )
```

Reply:
nop sapphire doesnt do space between control statement and parentheses